daemon/daemon.go

```diff
@@ -461,47 +458,34 @@ func (d *Daemon) compileBase() error {
 	args[initArgIPv6NodeIP] = nodeaddress.GetIPv6().String()
 	args[initArgIPv4ServiceRange] = v4ServicePrefix
 	args[initArgIPv6ServiceRange] = v6ServicePrefix
+	args[initArgTunnelMode] = tunnelMode
+	args[initArgDevice] = device
 
-	if d.conf.Device != "undefined" {
-		_, err := netlink.LinkByName(d.conf.Device)
+	if device != "undefined" {
+		_, err := netlink.LinkByName(device)
 		if err != nil {
-			log.Warningf("Link %s does not exist: %s", d.conf.Device, err)
+			log.Warningf("Link %s does not exist: %s", device, err)
 			return err
 		}
 
 		if d.conf.IsLBEnabled() {
-			if d.conf.Device != d.conf.LBInterface {
+			if device != d.conf.LBInterface {
 				//FIXME: allow different interfaces
 				return fmt.Errorf("Unable to have an interface for LB mode different than snooping interface")
 			}
 			if err := d.setHostAddresses(); err != nil {
 				return err
 			}
-			mode = "lb"
-		} else {
-			mode = "direct"
 		}
 
 		// in direct routing mode, only packets to the local node
 		// prefix should go to cilium_host
 		args[initArgIPv4Range] = nodeaddress.GetIPv4AllocRange().String()
 		args[initArgIPv6Range] = nodeaddress.GetIPv6NodeRange().String()
-
-		args[initArgMode] = mode
-		args[initArgDevice] = d.conf.Device
-
-		args = append(args, d.conf.Device)
 	} else {
-		if d.conf.IsLBEnabled() {
```

Comment:
Yes, LB is part of bpf_netdev and thus always available in direct mode as well. What we don't support yet is a multi interface environment.